src/layer/AbstractLayer.ts

```diff
@@ -89,8 +89,11 @@ export class AbstractLayer {
           'Method getHugeMap() requests that width and height of resulting image are specified',
         );
       }
+      if (params.format !== MimeTypes.JPEG || params.format !== MimeTypes.PNG) {
+        throw new Error('getHugeMap does not support' + params.format + 'only PNG and JPEG are supported');
+      }
```

Comment:
- `||` should be `&&`
- spaces are missing in the error message (`...does not supportTIFFonly PNG and JPEG...`).

I would suggest:
```
      if (params.format !== MimeTypes.JPEG && params.format !== MimeTypes.PNG) {
        throw new Error('getHugeMap() only supports PNG and JPEG image formats');
      }
```

Reply:
I see you changed the error message, but it is a bit misleading... `params.format` is supposed to be a MIME string, while `PNG` and `JPEG` are not... One could get such error message:
`getHugeMap does not support PNG only PNG and JPEG are supported`.

Also, we don't actually write the function name in other error messages, so we should probably remove `getHugeMap` for consistency.

The simplest solution would be to not print the `params.format`, but if you do, `PNG` and `JPEG` should be their MIME strings:
```javascript
      if (params.format !== MimeTypes.JPEG && params.format !== MimeTypes.PNG) {
        throw new Error('Format ' + params.format + ' not supported, only ' + MimeTypes.PNG + ' and ' + MimeTypes.JPEG + ' are allowed');
      }
```

Reply:
Changed it to what you suggested